examples/reshare/src/orchestrator/actor.rs

```diff
@@ -162,53 +170,127 @@ where
         );
         mux.start();
 
+        // Fuse the backup streams
+        let mut backup = futures::stream::select(pending_backup, recovered_backup);
+
         // Wait for instructions to transition epochs.
         let mut engines = BTreeMap::new();
-        while let Some(transition) = self.mailbox.next().await {
-            match transition {
-                Message::Enter(transition) => {
-                    // If the epoch is already in the map, ignore.
-                    if engines.contains_key(&transition.epoch) {
+        loop {
+            select! {
+                message = backup.next() => {
+                    // If a message is received in an unregistered sub-channel in the recovered or pending network,
+                    // attempt to forward the boundary finalization for the epoch.
+
+                    let Some((epoch, (from, mut message))) = message else {
+                        warn!("recovered/pending mux backup channel closed, shutting down orchestrator");
+                        break;
+                    };
+
+                    let Some(latest_epoch) = engines.keys().last().copied() else {
+                        debug!(epoch, ?from, "received message from unregistered epoch with no known epochs");
+                        continue;
+                    };
+
+                    if epoch as u64 > latest_epoch {
+                        // The sender is operating in a newer epoch, ignore. We cannot block them,
+                        // since they may be validly ahead.
+                        debug!(epoch, ?from, "received message from future epoch");
                         continue;
+                    } else if epoch as u64 == latest_epoch {
+                        panic!("recovered/pending mux backup channel received message for current epoch");
```

Comment:
This could validly occur due to async message processing (a message could be stuck on one of these backup channels when we update our latest epoch).

We should just drop (not panic).